composer.json

```diff
@@ -12,18 +12,14 @@
     "prefer-stable": true,
     "minimum-stability": "dev",
     "require": {
-        "php": "^5.3 | ^7.0",
-        "dhii/exception-interface": "^0.1 | ^0.2",
-        "dhii/data-key-value-aware-interface": "^0.1",
-        "dhii/factory-interface": "^0.1-alpha1"
+        "php": "^7.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.8",
-        "ptrofimov/xpmock": "^1.1",
+        "phpunit/phpunit": "^6.0",
         "dhii/php-cs-fixer-config": "dev-php-5.3",
         "codeclimate/php-test-reporter": "<=0.3.2",
-        "dhii/stringable-interface": "^0.1",
-        "psr/container": "^1.0"
+        "psr/container": "^1.0",
+        "container-interop/service-provider": "^0.4"
```

Comment:
I've been thinking about interface dependencies lately. I know that you are a proponent of having these dependencies be declared as dev dependencies since they are not required by the code in this repository.

But I realized that this means that later on, a consumer package can requires this module packlage can also require a container or service provider package is not compatible. For instance, `psr/container:2.0` or `container-interop/service-provider:3.0`, etc. There's a possibility that these packages no longer provide the interfaces that are loosely referenced in this package (like `ContainerInterface`).

As such, I'd like us to reconsider this practice and discuss whether we should start moving these interface dependencies to be normal required dependencies.

Reply:
E.g. `psr/container` can break BC, no problem. By dev-depending on `^1`, we are testing on that version and not on other versions. So, we do not guarantee compatibility with things we don't test with.

Of course, the use won't see a conflict reported by Composer. But this doesn't sound very important to me.

Reply:
That sounds very important to me. If I install a package that requires another but doesn't specify, Composer won't complain but my application won't run. I'd need to spend time figuring out what I need to install.

The same applies here IMO. Consumers can either not install or install a version that is not compatible, and Composer won't complain.

Reply:
Ok, I agree.